File: bindings/kotlin/src/main/kotlin/at/schem/nucleation/FieldProgramDistanceKind.kt
package at.schem.nucleation

import com.sun.jna.Callback
import com.sun.jna.Library
import com.sun.jna.Native
import com.sun.jna.Pointer
import com.sun.jna.Structure

internal interface FieldProgramDistanceKindLib: Library {
}
/** What kind of distance a field program's output represents.
*/
enum class FieldProgramDistanceKind {
    Exact,
    LowerBound,
    Estimate,
    Implicit;

    fun toNative(): Int {
        return this.ordinal
    }


    companion object {
        internal val libClass: Class<FieldProgramDistanceKindLib> = FieldProgramDistanceKindLib::class.java
        internal val lib: FieldProgramDistanceKindLib = Native.load("nucleation", libClass)
        fun fromNative(native: Int): FieldProgramDistanceKind {
            return FieldProgramDistanceKind.entries[native]
        }

        fun default(): FieldProgramDistanceKind {
            return Exact
        }
    }
}

File: bindings/php/FieldProgramDistanceKind.php
<?php
namespace Stencil;

final class FieldProgramDistanceKind {
    const Exact = 0;
    const LowerBound = 1;
    const Estimate = 2;
    const Implicit = 3;

    public static function name(int $value): string {
        return match ($value) {
            self::Exact => 'Exact',
            self::LowerBound => 'LowerBound',
            self::Estimate => 'Estimate',
            self::Implicit => 'Implicit',
            default => "Unknown({$value})",
        };
    }
}
